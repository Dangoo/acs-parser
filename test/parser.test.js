const acsParser = require('../index');

it('Parses null-values correctly', () => {
    const nullValues = [
        'key1 ',
        'key2 '
    ].concat('\n').join('\n');

    const parsed = acsParser.parse(nullValues);

    expect(parsed).toMatchSnapshot();
});

it('Parses string-values correctly', () => {
    const stringValues = [
        'key1 "value"',
        'key2 "bar"'
    ].concat('\n').join('\n');

    const parsed = acsParser.parse(stringValues);

    expect(parsed).toMatchSnapshot();
});

it('Parses numeric-values correctly', () => {
    const numericValues = [
        'key1 1',
        'key2 2',
        'key3 2.5',
        'key4 20',
        'key5 12345.123456'
    ].concat('\n').join('\n');

    const parsed = acsParser.parse(numericValues);

    expect(parsed).toMatchSnapshot();
});

it('Parses numeric-array-values correctly', () => {
    const numericArrayValues = [
        'key1 1,2',
        'key2 2, 4',
        'key3 10, 20, 30, 40'
    ].concat('\n').join('\n');

    const parsed = acsParser.parse(numericArrayValues);

    expect(parsed).toMatchSnapshot();
});

it('Parses kuid-values correctly', () => {
    const numericArrayValues = [
        'kuid <kuid:123456:123456>',
        'kuid-uppercase <KUID:123456:123456>',
        'kuid2 <kuid2:123456:123456:1>',
        'kuid2-uppercase <KUID2:123456:123456:1>',
    ].concat('\n').join('\n');

    const parsed = acsParser.parse(numericArrayValues);

    expect(parsed).toMatchSnapshot();
});

it('Parses container-values correctly', () => {
    const containerValues = [
        'container-value ',
        '{',
        '  nested-container-value ',
        '  {',
        '    nested-string "Nested"',
        '  }',
        '}'
    ].concat('\n').join('\n');

    const parsed = acsParser.parse(containerValues);

    expect(parsed).toMatchSnapshot();
});

it('Parses key-value-pairs correctly', () => {
    const numericArrayValues = [
        'kuid <kuid:123456:123456>',
        'string "Test"',
        'number 2',
        'container-value ',
        '{',
        '  nested-string "Nested"',
        '}'
    ].concat('\n').join('\n');

    const parsed = acsParser.parse(numericArrayValues);

    expect(parsed).toMatchSnapshot();
});