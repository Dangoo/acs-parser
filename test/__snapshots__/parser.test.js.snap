// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parses container-values correctly 1`] = `
Object {
  "container-value": Object {
    "type": "container",
    "value": Object {
      "nested-container-value": Object {
        "type": "container",
        "value": Object {
          "nested-string": Object {
            "type": "string",
            "value": "Nested",
          },
        },
      },
    },
  },
}
`;

exports[`Parses key-value-pairs correctly 1`] = `
Object {
  "container-value": Object {
    "type": "container",
    "value": Object {
      "nested-string": Object {
        "type": "string",
        "value": "Nested",
      },
    },
  },
  "kuid": Object {
    "type": "kuid",
    "value": Object {
      "contentID": "123456",
      "userID": "123456",
    },
  },
  "number": Object {
    "type": "number",
    "value": 2,
  },
  "string": Object {
    "type": "string",
    "value": "Test",
  },
}
`;

exports[`Parses kuid-values correctly 1`] = `
Object {
  "kuid": Object {
    "type": "kuid",
    "value": Object {
      "contentID": "123456",
      "userID": "123456",
    },
  },
  "kuid-uppercase": Object {
    "type": "kuid",
    "value": Object {
      "contentID": "123456",
      "userID": "123456",
    },
  },
  "kuid2": Object {
    "type": "kuid2",
    "value": Object {
      "contentID": "123456",
      "userID": "123456",
      "version": 1,
    },
  },
  "kuid2-uppercase": Object {
    "type": "kuid2",
    "value": Object {
      "contentID": "123456",
      "userID": "123456",
      "version": 1,
    },
  },
}
`;

exports[`Parses null-values correctly 1`] = `
Object {
  "key1": null,
  "key2": null,
}
`;

exports[`Parses numeric-array-values correctly 1`] = `
Object {
  "key1": Object {
    "type": "numeric-array",
    "value": Array [
      1,
      2,
    ],
  },
  "key2": Object {
    "type": "numeric-array",
    "value": Array [
      2,
      4,
    ],
  },
  "key3": Object {
    "type": "numeric-array",
    "value": Array [
      10,
      20,
      30,
      40,
    ],
  },
}
`;

exports[`Parses numeric-values correctly 1`] = `
Object {
  "key1": Object {
    "type": "number",
    "value": 1,
  },
  "key2": Object {
    "type": "number",
    "value": 2,
  },
  "key3": Object {
    "type": "number",
    "value": 2.5,
  },
  "key4": Object {
    "type": "number",
    "value": 20,
  },
  "key5": Object {
    "type": "number",
    "value": 12345.123456,
  },
}
`;

exports[`Parses string-values correctly 1`] = `
Object {
  "key": Object {
    "type": "string",
    "value": "new
        
        line",
  },
  "key1": Object {
    "type": "string",
    "value": "value",
  },
  "key2": Object {
    "type": "string",
    "value": "bar",
  },
  "key3": Object {
    "type": "string",
    "value": "öäüßÖÄÜøØ",
  },
}
`;
